src/main/java/org/junit/internal/runners/ErrorReportingRunner.java

```diff
@@ -13,19 +13,39 @@
 public class ErrorReportingRunner extends Runner {
     private final List<Throwable> causes;
 
+    // Only one of the following two fields will be non-null.
     private final Class<?> testClass;
+    private final String displayName;
 
     public ErrorReportingRunner(Class<?> testClass, Throwable cause) {
         if (testClass == null) {
             throw new NullPointerException("Test class cannot be null");
         }
         this.testClass = testClass;
+        displayName = null;
+        causes = getCauses(cause);
+    }
+
+    /**
+     * @since 4.13
+     */
+    public ErrorReportingRunner(String displayName, Throwable cause) {
+        if (displayName == null) {
+            throw new NullPointerException("Display name cannot be null");
+        }
+        testClass = null;
+        this.displayName = displayName;
         causes = getCauses(cause);
```

Comment:
While that would lead to a (slight) decrease in code duplication, I think it would sacrifice clarity. It is extremely clear what the constructors do